diff --git a/core/fxcrt/BUILD.gn b/core/fxcrt/BUILD.gn
index f80726b..778e4a4 100644
--- a/core/fxcrt/BUILD.gn
+++ b/core/fxcrt/BUILD.gn
@@ -171,7 +171,7 @@ source_set("fxcrt") {
   } else {
     sources += [ "fx_memory_malloc.cpp" ]
   }
-  if (is_posix) {
+  if (is_posix || is_wasm) {
     sources += [
       "cfx_fileaccess_posix.cpp",
       "cfx_fileaccess_posix.h",
diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index 4743d22..51dc615 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -171,7 +171,7 @@ source_set("fxge") {
     ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_wasm) {
     sources += [ "linux/fx_linux_impl.cpp" ]
   }
 
diff --git a/core/fxge/cfx_face.cpp b/core/fxge/cfx_face.cpp
index a2895d7..2dd0db3 100644
--- a/core/fxge/cfx_face.cpp
+++ b/core/fxge/cfx_face.cpp
@@ -1142,10 +1142,7 @@ std::unique_ptr<CFX_GlyphBitmap> CFX_Face::RenderGlyph(
 
   if (!loaded_fontations_outline) {
     ScopedFaceTransform scoped_transform(GetRec(), &ft_matrix);
-    int load_flags = FT_LOAD_NO_BITMAP | FT_LOAD_PEDANTIC;
-    if (!IsTtOt()) {
-      load_flags |= FT_LOAD_NO_HINTING;
-    }
+    int load_flags = FT_LOAD_NO_BITMAP | FT_LOAD_PEDANTIC | FT_LOAD_NO_HINTING;
     int error = FT_Load_Glyph(rec, glyph_index, load_flags);
     if (error) {
       if (load_flags & FT_LOAD_NO_HINTING) {
