diff --git a/core/fxcrt/binary_buffer_unittest.cpp b/core/fxcrt/binary_buffer_unittest.cpp
index 30a7e24..eea5215 100644
--- a/core/fxcrt/binary_buffer_unittest.cpp
+++ b/core/fxcrt/binary_buffer_unittest.cpp
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "core/fxcrt/binary_buffer.h"
+#include "core/fxcrt/byteorder.h"
 
 #include <utility>
 #include <vector>
@@ -122,7 +123,7 @@ TEST(BinaryBuffer, AppendBytes) {
 // Assumes little endian.
 TEST(BinaryBuffer, AppendUint16) {
   BinaryBuffer buffer;
-  buffer.AppendUint16(0x4321);
+  buffer.AppendUint16(FromLE16(0x4321));
   EXPECT_EQ(2u, buffer.GetSize());
   EXPECT_EQ(2u, buffer.GetLength());
   EXPECT_EQ(0x21u, buffer.GetSpan()[0]);
@@ -132,7 +133,7 @@ TEST(BinaryBuffer, AppendUint16) {
 // Assumes little endian.
 TEST(BinaryBuffer, AppendUint32) {
   BinaryBuffer buffer;
-  buffer.AppendUint32(0x87654321);
+  buffer.AppendUint32(FromLE32(0x87654321));
   EXPECT_EQ(4u, buffer.GetSize());
   EXPECT_EQ(4u, buffer.GetLength());
   EXPECT_EQ(0x21u, buffer.GetSpan()[0]);
diff --git a/core/fxcrt/cfx_seekablestreamproxy.cpp b/core/fxcrt/cfx_seekablestreamproxy.cpp
index f0b4143..b6e1355 100644
--- a/core/fxcrt/cfx_seekablestreamproxy.cpp
+++ b/core/fxcrt/cfx_seekablestreamproxy.cpp
@@ -91,11 +91,19 @@ void SwapByteOrder(pdfium::span<uint16_t> str) {
 // Returns the code page indicated by a byte-order mark at the start of a
 // stream, or nullopt if no BOM is present.
 std::optional<FX_CodePage> DetectByteOrderMark(uint32_t leading_bytes) {
+#if defined(ARCH_CPU_LITTLE_ENDIAN)
   constexpr uint32_t kUtf8Mask = 0x00FFFFFF;
   constexpr uint32_t kUtf8Marker = 0x00BFBBEF;
   constexpr uint32_t kUtf16Mask = 0x0000FFFF;
   constexpr uint32_t kUtf16BeMarker = 0x0000FFFE;
   constexpr uint32_t kUtf16LeMarker = 0x0000FEFF;
+#else
+  constexpr uint32_t kUtf8Mask = 0xFFFFFF00;
+  constexpr uint32_t kUtf8Marker = 0xEFBBBF00;
+  constexpr uint32_t kUtf16Mask = 0xFFFF0000;
+  constexpr uint32_t kUtf16BeMarker = 0xFEFF0000;
+  constexpr uint32_t kUtf16LeMarker = 0xFFFE0000;
+#endif
 
   if ((leading_bytes & kUtf8Mask) == kUtf8Marker) {
     return FX_CodePage::kUTF8;
@@ -196,9 +204,15 @@ size_t CFX_SeekableStreamProxy::ReadBlock(pdfium::span<wchar_t> buffer) {
     size_t bytes_read =
         ReadData(pdfium::as_writable_bytes(buffer).first(bytes_to_read));
     size_t elements = bytes_read / sizeof(uint16_t);
+#if defined(ARCH_CPU_LITTLE_ENDIAN)
     if (code_page_ == FX_CodePage::kUTF16BE) {
       SwapByteOrder(fxcrt::reinterpret_span<uint16_t>(buffer).first(elements));
     }
+#else
+    if (code_page_ == FX_CodePage::kUTF16LE) {
+      SwapByteOrder(fxcrt::reinterpret_span<uint16_t>(buffer).first(elements));
+    }
+#endif
     UTF16ToWChar(buffer.first(elements));
     return elements;
   }
