diff --git a/third_party/libc++/BUILD.gn b/third_party/libc++/BUILD.gn
index 8c61827..18cb890 100644
--- a/third_party/libc++/BUILD.gn
+++ b/third_party/libc++/BUILD.gn
@@ -44,10 +44,10 @@ config("stdver") {
       # We want to use a uniform C++ version across all of chromium, but
       # upstream libc++ requires C++23 so we have to make an exception here.
       # No other target should override the default -std= flag.
-      "-std:c++23preview",
+      "-std:c++20",  # XXX(pypdfium2)
     ]
   } else {
-    cflags_cc = [ "-std=c++23" ]
+    cflags_cc = [ "-std=c++20" ]  # XXX(pypdfium2)
   }
 }
 
