diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 12595c1..64be83c 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -371,8 +371,8 @@ default_compiler_configs = [
   "//build/config/compiler:no_rtti",
   "//build/config/compiler:no_unresolved_symbols",
   "//build/config/compiler:runtime_library",
-  "//build/config/compiler:sanitize_c_array_bounds",
-  "//build/config/compiler:sanitize_return",
+  # "//build/config/compiler:sanitize_c_array_bounds",
+  # "//build/config/compiler:sanitize_return",
   "//build/config/compiler:thin_archive",
   "//build/config/compiler:thinlto_optimize_default",
   "//build/config/compiler:tot_warnings",
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index e1da114..8aed18c 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -576,12 +576,6 @@ config("compiler") {
   if (is_clang) {
     # Flags for diagnostics.
     cflags += [ "-fcolor-diagnostics" ]
-    if (!is_win) {
-      cflags += [ "-fdiagnostics-show-inlining-chain" ]
-    } else {
-      # Combine after https://github.com/llvm/llvm-project/pull/192241
-      cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
-    }
     if (diagnostics_print_source_range_info && !is_win) {
       cflags += [ "-fdiagnostics-print-source-range-info" ]
     }
@@ -612,13 +606,6 @@ config("compiler") {
       ]
     }
 
-    # The performance improvement does not seem worth the risk. See
-    # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
-    # for discussion.
-    if (!is_wasm) {
-      cflags += [ "-fno-lifetime-dse" ]
-    }
-
     # TODO(hans): Remove this once Clang generates better optimized debug info
     # by default. https://crbug.com/765793
     cflags += [
@@ -1922,7 +1909,7 @@ config("default_warnings") {
       "-Wno-uninitialized-const-pointer",
 
       # TODO(crbug.com/495753203): Fix and re-enable.
-      "-Wno-unused-but-set-global",
+      # "-Wno-unused-but-set-global",
     ]
 
     cflags_cc += [
