diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index 1f434d4..cb9fc1f 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -407,8 +407,8 @@ template("single_gcc_toolchain") {
         command = "cmd /s /c \"\"$python_path\" $tool_wrapper_path delete-file {{output}} && $command\""
       } else {
         command = "rm -f {{output}} && $command"
-        inputs =
-            [ get_path_info(rebase_path(ar, ".", root_out_dir), "abspath") ]
+        # inputs =
+        #    [ get_path_info(rebase_path(ar, ".", root_out_dir), "abspath") ]
       }
 
       # Almost all targets build with //build/config/compiler:thin_archive which
