diff --git a/build/gen.py b/build/gen.py
index 5d161e4..5422e5d 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -239,8 +239,9 @@ def is_gcc(cxx):
       f'{cxx} -dM -E -',
       shell=True,
       stdin=subprocess.DEVNULL,
-      text=True,
-      capture_output=True)
+      universal_newlines=True,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.STDOUT)
 
   return ret.returncode == 0 and "#define __GNUC__" in ret.stdout and not "#define __clang__" in ret.stdout
 
diff --git a/out/last_commit_position.h b/out/last_commit_position.h
new file mode 100644
index 0000000..5e6b8e4
--- /dev/null
+++ b/out/last_commit_position.h
@@ -0,0 +1,7 @@
+#ifndef OUT_LAST_COMMIT_POSITION_H_
+#define OUT_LAST_COMMIT_POSITION_H_
+
+#define LAST_COMMIT_POSITION_NUM 2289
+#define LAST_COMMIT_POSITION "2289 (a0c5124a5060)"
+
+#endif  // OUT_LAST_COMMIT_POSITION_H_
