diff --git a/build/gen.py b/build/gen.py
index 432924e4..5ce59771 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -239,8 +239,9 @@ def is_gcc(cxx):
       f'{cxx} -dM -E -',
       shell=True,
       stdin=subprocess.DEVNULL,
-      text=True,
-      capture_output=True)
+      universal_newlines=True,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.STDOUT)
 
   return ret.returncode == 0 and "#define __GNUC__" in ret.stdout and not "#define __clang__" in ret.stdout
 
diff --git a/out/last_commit_position.h b/out/last_commit_position.h
new file mode 100644
index 00000000..e24cea78
--- /dev/null
+++ b/out/last_commit_position.h
@@ -0,0 +1,7 @@
+#ifndef OUT_LAST_COMMIT_POSITION_H_
+#define OUT_LAST_COMMIT_POSITION_H_
+
+#define LAST_COMMIT_POSITION_NUM 2385
+#define LAST_COMMIT_POSITION "2385 (9ece3f5254c2)"
+
+#endif  // OUT_LAST_COMMIT_POSITION_H_
