diff --git a/BUILD.gn b/BUILD.gn
index 25831ae..4708944 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -94,11 +94,17 @@
       "mips/filter_msa_intrinsics.c",
       "mips/mips_init.c",
     ]
-  } else if (current_cpu == "loong64" && loongarch64_use_lsx) {
-    sources += [
-      "loongarch/filter_lsx_intrinsics.c",
-      "loongarch/loongarch_lsx_init.c",
-    ]
+  } else if (current_cpu == "loong64") {
+    if (loongarch64_use_lsx) {
+      sources += [
+        "loongarch/filter_lsx_intrinsics.c",
+        "loongarch/loongarch_lsx_init.c",
+      ]
+    } else {
+      defines += ["PNG_LOONGARCH_LSX_OPT=0"]
+    }
+  } else if (current_cpu == "ppc64") {
+    defines += ["PNG_POWERPC_VSX_OPT=0"]
   }
 
   configs -= [ "//build/config/compiler:chromium_code" ]
