diff --git a/config/clang/BUILD.gn b/config/clang/BUILD.gn
index e05032c..5e6229c 100644
--- a/config/clang/BUILD.gn
+++ b/config/clang/BUILD.gn
@@ -204,6 +204,9 @@ template("clang_lib") {
         } else {
           assert(false)  # Unhandled cpu type
         }
+        _tmp_split = string_split(_dir, "-")
+        _suffix = "-" + _tmp_split[0]
+        _dir = "linux"
       } else if (is_fuchsia) {
         if (current_cpu == "x64") {
           _dir = "x86_64-unknown-fuchsia"
