diff --git a/BUILD.gn b/BUILD.gn
index 4edba9f43..169cbeb1a 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -219,7 +219,7 @@ group("pdfium_public_headers") {
   ]
 }
 
-component("pdfium") {
+shared_library("pdfium") {
   output_name = "pdfium"
   libs = []
   configs += [ ":pdfium_strict_config" ]
diff --git a/public/fpdfview.h b/public/fpdfview.h
index 6922da991..86f327016 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -203,7 +203,7 @@ typedef int FPDF_ANNOT_APPEARANCEMODE;
 // Dictionary value types.
 typedef int FPDF_OBJECT_TYPE;
 
-#if defined(COMPONENT_BUILD)
+#if 1  // defined(COMPONENT_BUILD)
 // FPDF_EXPORT should be consistent with |export| in the pdfium_fuzzer
 // template in testing/fuzzers/BUILD.gn.
 #if defined(WIN32)
