--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -557,8 +557,7 @@ foreach(_target_type,
         deps = []
       }
 
-      if (is_clang && (!defined(invoker.use_libcxx_modules) ||
-                       invoker.use_libcxx_modules)) {
+      if (is_clang && false) {
         # This is necessary for Clang modules builds.
         deps += [
           "//buildtools/third_party/libc++:std",
@@ -661,8 +660,7 @@ foreach(_target_type,
         deps += [ "//build/config:${_target_type}_deps" ]
       }
 
-      if (is_clang && (!defined(invoker.use_libcxx_modules) ||
-                       invoker.use_libcxx_modules)) {
+      if (is_clang && false) {
         # These are necessary for Clang modules builds.
         deps += [
           "//buildtools/third_party/libc++:_Builtin_limits",
