--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -593,20 +593,20 @@ config("compiler") {
     # TODO(crbug.com/40283598): This causes binary size growth and potentially
     # other problems.
     # TODO(crbug.com/40284925): This isn't supported by Cronet's mainline llvm version.
-    if (default_toolchain != "//build/toolchain/cros:target" &&
-        !llvm_android_mainline) {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
-      if (use_thin_lto && is_a_target_toolchain) {
-        if (is_win) {
-          ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
-        } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
-        }
-      }
-    }
+    # if (default_toolchain != "//build/toolchain/cros:target" &&
+    #     !llvm_android_mainline) {
+    #   cflags += [
+    #     "-mllvm",
+    #     "-split-threshold-for-reg-with-hint=0",
+    #   ]
+    #   if (use_thin_lto && is_a_target_toolchain) {
+    #     if (is_win) {
+    #       ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
+    #     } else {
+    #       ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
+    #     }
+    #   }
+    # }
 
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
@@ -617,10 +617,10 @@ config("compiler") {
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF
     # (excluding toolchains that use an older version of LLVM).
-    if (is_linux && !llvm_android_mainline &&
-        default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
-    }
+    # if (is_linux && !llvm_android_mainline &&
+    #     default_toolchain != "//build/toolchain/cros:target") {
+    #   cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
+    # }
   }
 
   # C11/C++11 compiler flags setup.
@@ -1984,10 +1984,10 @@ config("default_warnings") {
         "-Wno-cast-function-type",
 
         # TODO(crbug.com/40284799): Fix and re-enable.
-        "-Wno-thread-safety-reference-return",
+        # "-Wno-thread-safety-reference-return",
 
         # TODO(crbug.com/376641662): Fix and re-enable.
-        "-Wno-nontrivial-memcall",
+        # "-Wno-nontrivial-memcall",
       ]
 
       cflags_cc += [
